subsys/testsuite/ztest/src/ztest_new.c

```diff
@@ -21,6 +21,18 @@ static struct k_thread ztest_thread;
 #include <unistd.h>
 #endif
 
+#if (IS_ENABLED(CONFIG_ZTEST_DO_THE_SHUFFLE))
```

Comment:
nit: I think the general convention is to use `IS_ENABLED` inside a logic block and at the top level use `#ifdef`. Sorry about the confusion.